mods/page_revisions.js

```diff
@@ -292,8 +302,46 @@ PRS.prototype.getTitleRevision = function(restbase, req) {
             }
             return self.fetchAndStoreMWRevision(restbase, req);
         });
-    } else if (!rp.revision) {
-        revisionRequest = self.fetchAndStoreMWRevision(restbase, req);
+    } else if (!rp.revision || rp.revision === 'latest') {
+        revisionRequest = self.fetchAndStoreMWRevision(restbase, req)
+            .catch(function (e) {
+                if (e.status !== 404) {
+                    throw e;
+                }
+                // In case 404 is returned by MW api, the page is deleted
+                return self.listTitleRevisions(restbase, req)
+                    .then(function (res) {
+                        if (res.body.items && res.body.items.length > 0 ) {
+                            var revReq = {
+                                uri: new URI([rp.domain, 'sys', 'page_revisions', 'rev', res.body.items[0]]),
+                                params: {
```

Comment:
The `params` object attribute here is superfluous - URI parameters are extracted from the URI automatically by RESTBase already.